gammapy/spectrum/models.py

```diff
@@ -660,10 +664,38 @@ def read_xspec_model(cls, filename, param):
         return cls(energy=energy, values=values, scale_logy=False)
 
     def evaluate(self, energy, amplitude):
-        interpy = self.interpy(np.log10(energy.to('eV').value))
+
+        is_array = True
```

Comment:
Can we try to avoid the if - else for different inputs as much as possible?

As far as I can see, all the other model evaluate functions don't do something like this.
Can't you just assume inputs are Quantity?

If this is needed for some reason, please convert to a standard (e.g. quantity or numpy array) at the top, and then the rest of the method is without if / else, just processing the data in a standard format.